src/App.tsx

```diff
@@ -1,14 +1,55 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { FilterTypes } from './types/Filter';
 
 export const App: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filterBy, setFilterBy] = useState(FilterTypes.All);
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    const downloadData = async () => {
+      const data = await getTodos()
+        .then(todo => {
+          setTodos(todo);
+          setIsLoading(false);
+        });
+
+      return data;
+    };
+
+    downloadData();
+  }, []);
+
+  let filterTodos = todos
+    .filter(todoItem => {
+      switch (filterBy) {
+        case FilterTypes.Active:
+          return !todoItem.completed;
+
+        case FilterTypes.Completed:
+          return todoItem.completed;
+
+        default:
+          return todoItem;
+      }
+    });
+
+  if (query) {
+    filterTodos = filterTodos.filter(todoItem => (
+      todoItem.title.toLowerCase().includes(query.toLowerCase())));
+  }
```

Comment:
you still filter two times )
you can create method for check includes `checkContainsQueryInTitle` and use it on your filter:
```
const visibleTodos = todos
    .filter(todo => {
      switch (selectType) {
        case 'active':
          return !todo.completed && checkContainsQueryInTitle(todo.title);

        case 'completed':
          return todo.completed && checkContainsQueryInTitle(todo.title);

        default:
          return checkContainsQueryInTitle(todo.title);
      }
    })
```